### ISSUE TITLE
Add Support for Process's ParentProcessId on FreeBSD

### ISSUE THREAD

**Issue by bgribaudo** [dotnet/corefx#35290]
### Request/Need
Implement  private instance method `Process.ParentProcessId` method on FreeBSD. Presumably this will involve adding an interop method that calls the appropriate native API method.

### Background
Currently, this method is not supported on FreeBSD (e.g. throws a `PlatformNotSupportedException`).  This lack blocks `Kill(entireProcessTree: true)` from being supported on FreeBSD.

Related: https://github.com/dotnet/corefx/pull/34147#discussion_r248261306

**Comment by wfurt** [comment]
do you have simple code to run to reproduce this @bgribaudo ?

**Comment by bgribaudo** [comment]
Off the top of my head, maybe just try calling `process.Kill(entireProcessTree: true)` on a process instance?

**Comment by am11** [comment]
I think this FreeBSD implementation will fill the gap: https://github.com/am11/corefx/commit/fe9254b980ae777a0d7c08b6290b1026259cec86.

I didn't open a pull request due to the blocker that we are unable to build CoreFX on FreeBSD; hitting package restore errors (https://github.com/dotnet/corefx/issues/2046#issuecomment-481739054). If building CoreFX via source-build is a viable option with <kbd>release/3.0</kbd> branch, I can attempt @wfurt, @dagood (on a private VM for now).

Otherwise, we can park it until the whole `azure-pipelines-agent -> core-setup -> core-sdk -> corefx,coreclr` chain is unblocked. 😢

**Comment by bgribaudo** [comment]
Thanks for working on this, @am11! It will be nice to see this supported on another platform.

**Comment by wfurt** [comment]
I would probably add ppid to proc_stats instead of dragging unsafe code to Process.cs. I'll see what I can do about the build. It would be nice to have it back.

**Comment by am11** [comment]
Agree, your approach is much cleaner @wfurt: https://github.com/am11/corefx/commit/871cf28?w=1
(remove `?w=1` to compare with whitespace/formatting changes)

**Comment by am11** [comment]
The property was implemented in master branch, can be closed.

**Comment by danmosemsft** [comment]
Thanks @am11
